### PR TITLE
feat: allow flexible application status changes

### LINKED ISSUE #70: Allow Organizers to Update Applicant Status After Decision
Currently, once an applicant is marked as Accepted or Rejected, the decision cannot be changed. This makes it difficult for organizers to correct mistakes or update an applicant's status if circumstances change.
To improve flexibility, the application review workflow could follow the same approach as the Speaker Review system by using a status dropdown.

Image

**Expected Behavior**
- Allow organizers to change an applicant's status after the initial review.
- Use a status dropdown, similar to the Speaker Review workflow.

Image

- Organizers should be able to update the status whenever necessary (e.g., to correct an accidental selection or revise a decision).

### PR DESCRIPTION
**Pull request by MukundC25**
#### Summary
This PR introduces a flexible application review workflow, allowing organizers to change a volunteer applicant's status at any point after the initial review. 

Previously, the workflow relied on a static label after an applicant was marked as Accepted or Rejected. This update replaces that label with a dynamic `<select>` dropdown in the Applications list view, enabling seamless status corrections (e.g., reverting an accidental rejection or updating an accepted applicant).

**Changes:**
* **View Updates:** Refactored `ApplicationStatusView` to accept dynamic `status` payloads via POST, rather than hardcoded `action=accept|reject` kwargs.
* **UI Implementation:** Implemented the dropdown using a native HTML `<select>` element styled via semantic classes in `teamshifts.css`. This completely circumvents the clipping and wrapping bugs inherent to Bootstrap 3 `.btn-group` dropdowns when placed inside `.table-responsive` containers.
* **Test Coverage:** Updated test suites (e.g., `test_email_lifecycle_transactions.py`) to reflect the new POST payload structure (`status=<ApplicationStatus>`) and added coverage for edge cases, such as no-op behavior when the status is unchanged and preventing lifecycle emails when transitioning back to `pending`.

#### Expected Behavior
* Organizers can update an applicant's status via a dropdown menu in the Applications list table.
* The backend correctly handles the transition, verifying valid `ApplicationStatus` values and ignoring duplicate state submissions.
* The frontend dropdown perfectly mimics the native `.label` styling and works flawlessly in responsive table layouts without clipping.

https://github.com/user-attachments/assets/09581214-44af-454f-9466-1adff39e1365

Closes #70

### CODE REVIEW

File: teamshifts/static/teamshifts/teamshifts.css [hunk 1]
```diff
@@ -66,3 +66,13 @@ label .optional {
...
+
+.table-responsive {
+    overflow: visible;
+}
+
+@media (max-width: 767px) {
+    .table-responsive {
+        overflow-x: auto;
+    }
+}
```
[the remark this answers is not in the document]
Comment (MukundC25) [reply teamshifts/static/teamshifts/teamshifts.css]: Standard CSS-only approach to prevent `.table-responsive` from clipping Bootstrap dropdowns

**Review by Saksham-Sirohi — COMMENTED** [review]
## Review

Solid direction — status dropdown + no-op / pending-reset email handling look correct, and CI is green.

**Notes / follow-ups**
1. **Detail page regression:** `application_detail.html` no longer exposes accept/reject (or the new dropdown). Status is read-only there while the list can change it. Either add the same `<select>` on detail (and pass `status_choices` in `ApplicationDetailView.get_context_data`), or keep a clear affordance linking back to the list.
2. **Invalid payload → `Http404`:** Prefer `400`/`HttpResponseBadRequest` (or a form validation error) when `status` is missing/invalid; 404 implies the application was not found.
3. **Re-transition emails:** Accept→reject→accept (and reject→accept) will re-queue lifecycle mail every time. Confirm that is intentional; if not, gate on “first transition into accepted/rejected” or add a “notify” checkbox.
4. **Inline `onchange`:** Prefer a tiny external ES module listener over `onchange="this.form.submit()"` to match Eventyay JS conventions.
5. **Tests:** Nice coverage for no-op and pending reset. Consider one case for `accepted → rejected` (and optionally reject→accept) so re-email behavior stays pinned.

Mergeable / CI green. Not blocking merge if the detail UX is an accepted follow-up.

File: teamshifts/templates/teamshifts/applications.html [hunk 2]
```diff
@@ -65,44 +65,34 @@ <h1>{% trans "Applications" %}</h1>
...
-                                {% elif app.status == 'accepted' %}
-                                    <span class="label label-success">{% trans "Accepted" %}</span>
-                                {% else %}
-                                    <span class="label label-danger">{% trans "Rejected" %}</span>
-                                {% endif %}
+                            <td class="text-nowrap">
+                                <form action="{% url 'plugins:teamshifts:application_status' organizer=request.organizer.slug event=request.event.slug pk=app.pk %}"
+                                      method="post" id="status-form-{{ app.pk }}" class="teamshifts-inline-form">
+                                    {% csrf_token %}
+                                    <select name="status" aria-label="{% trans 'Change application status' %}" class="form-control teamshifts-status-dropdown {% if app.status == 'pending' %}label-warning{% elif app.status == 'accepted' %}label-success{% else %}label-danger{% endif %}" onchange="this.form.submit()">
```
Comment (Aqil-Ahmad) [inline teamshifts/templates/teamshifts/applications.html]: An accidental click/scroll-over-select silently emails the applicant. Consider a confirm dialog

**Review by Aqil-Ahmad — CHANGES_REQUESTED** [review]
the detail view now removes the accept reject buttons that is intentional right? also fix the comment for dialog, the emails are sent directly in teamshifts so at least we can have a dialog for confirmation.

**Review by Saksham-Sirohi — COMMENTED** [review]
Copilot flagged the following:

No additional findings beyond the existing review thread.

**Review by Aqil-Ahmad — APPROVED** [review]
minor: You're using the browser's native confirmation dialog. If there is already a reusable confirmation dialog, it would be better to use that; otherwise, you can implement one.

**Review by Saksham-Sirohi — CHANGES_REQUESTED** [review]
Please follow on @Aqil-Ahmad suggestion seems legit

File: teamshifts/templates/teamshifts/applications.html
Comment (SxxAq) [inline teamshifts/templates/teamshifts/applications.html:1]: The confirmation dialog text says "This may send an email to the applicant".  but resetting to pending or submitting the same status doesn't send anything. Consider only showing the confirmation for accepted/rejected transitions, or rewording to something like "Are you sure? This will update the application status."

**Review by SxxAq — COMMENTED** [review]
The native dialog works fine functionally, but a modal would match Eventyay's design language better as suggested by @Aqil-Ahmad also.

**Review by Aqil-Ahmad — APPROVED** [review]

**Review by Saksham-Sirohi — APPROVED** [review]
